tests/std/tests/P1659R3_ranges_alg_starts_with/test.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <functional>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+struct instantiator {
+    static constexpr pair<int, int> haystack[]       = {{0, 42}, {2, 42}, {4, 42}};
+    static constexpr pair<int, int> short_haystack[] = {{0, 42}};
+    static constexpr pair<long, long> needle[]       = {{13, 0}, {13, 2}};
+    static constexpr pair<long, long> wrong_needle[] = {{13, 0}, {13, 3}};
+
+    template <ranges::input_range In1, ranges::input_range In2>
+    static constexpr void test() {
+        using ranges::begin, ranges::end, ranges::starts_with;

```

Comment:
I observe that the `ends_with` test uses `ranges::equal_to`, but this test doesn't (so it's using `std::equal_to`).